test_notebooks/time_variable_Deff.py

```diff
@@ -0,0 +1,89 @@
+"""
+A script for calculating Deff based on MSD data from t=0 to a user-selected
+timepoint or range of timepoints (in which case Deff is a geometric mean).
+"""
+from matplotlib import pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy
+from scipy import stats
+
+from bokeh.plotting import figure, show, gridplot
+import zipfile
+
+
+ALPHA = 1
+
+
+zf = zipfile.ZipFile('brain-diffusion_data.zip')
+file_handle = zf.open('Mean_Square_Displacement_Data.csv')
+msd = pd.read_csv(file_handle)
+# Rename columns without biological replicate number (this allows for
+# averaging of biological replicates later)
+msd.columns = msd.iloc[0]
+msd = msd.iloc[1:]
+# Create a list of column names without replicates
+columns = msd.columns
+columns = columns[1:67]
+columns2 = [ii for n, ii in enumerate(columns) if ii not in columns[:n]]
+
+
+"""
+The following function combines MSDs of three biological replicates
+(different brain tissue samples) using a geometric mean.
+"""
+def compute_geomean(df, ColumnName):
+    geomeans = [0]
+    # Grab the three MSD values, one timepoint at a time
+    for i in range(2, len(df)+1):
+        timepoint = df[ColumnName].ix[i]
+        bioreps = list(timepoint[[0,1,2]])
```

Comment:
What if my dataset that I feed in has only two columns?  This would also be a good place for an if statement to check for the number of columns in the dataset.  You could pop out an error message for it, and you could make a unit test to address it.